toolkit/tools/imagecustomizerapi/partitionsize.go

```diff
@@ -66,6 +67,13 @@ func (s *PartitionSize) UnmarshalJSON(data []byte) error {
 	return parseAndSetPartitionSize(stringValue, s)
 }
 
+func (PartitionSize) JSONSchema() *jsonschema.Schema {
```

Comment:
You'll probably want a `JSONSchema` function on each type that has a `UnmarshalYAML` function.